packages/agreement-process/src/services/agreementService.ts

```diff
@@ -345,6 +345,39 @@ export function agreementServiceBuilder(
         })
       );
 
+      return agreementId;
+    },
+    async archiveAgreement(
+      agreementId: string,
+      authData: AuthData
+    ): Promise<Agreement["id"]> {
+      logger.info(`Archiving agreement ${agreementId}`);
```

Comment:
For consistency with the other calls, you could create a dedicated function with the archiveAgreementLogic? Even if it's very short, use the same pattern.

(just a suggestion, feel free to ignore)

Reply:
Good catch this is the best practices to make services unit-teastable 👍🏻